### ISSUE TITLE
lock resource queue is not working.

### ISSUE THREAD

**Issue by simplyviks** [jenkinsci/lockable-resources-plugin#101]
i have 13 resources with label atResource, so the first 13 resources are locked and automation runs on all 13 of them but after that the wait doesnt happens it directly return null and automation is triggered on null resources, this was working fine in older release until we migrated recently to latest one.
//called 20 times 
lock(label: atResource, quantity: 1) {
             //each time lock a resource and run automation on it.
              runAutomation(env.resource0)
}

Note:after migration to latest one the step is changed as below as env.resource0 no more works
lock(label: atResource, quantity: 1,variable:"resource0") {
             //each time lock a resource and run automation on it.
              runAutomation(env.resource0)
}

**Comment by Enteee** [comment]
Plugin: 2.3

I have observed similar behavior. When using the lockable resource plugin like this: 
```
options {
  lock(label: 'resource', variable: 'TEST_VM_NAME', quantity: 1)
}  
```
sometimes this has the effect that `env.TEST_VM_NAME` is set to `null`. But I do not know what the conditions are that make this happen.

**Comment by sterys** [comment]
Se bug https://issues.jenkins-ci.org/browse/JENKINS-50176

**Comment by TobiX** [comment]
Should be fixed with release 2.7 (see #117)
